geoai/universat.py

```
@@ -0,0 +1,230 @@
+    ) -> torch.Tensor:
+        scale = scale or 1.0
+        img_norm = img.astype(np.float32) / scale
+        if mod in TIME_SERIES_MODALITIES and img_norm.ndim == 3:
+            img_norm = np.expand_dims(img_norm, axis=0)
+        return torch.tensor(img_norm, dtype=torch.float32)
+
+    def _process_sample(
+        self, sample: Dict[str, Any], mod: str, scale: Optional[float]
+    ) -> Tuple[torch.Tensor, Optional[torch.Tensor]]:
+        if (val := sample.get(mod)) is None:
+            raise ValueError(f"Missing modality '{mod}'")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)